aten/src/ATen/native/cuda/IndexKernel.cu

```diff
@@ -63,11 +65,31 @@ void gpu_index_kernel(TensorIteratorBase& iter, const IntArrayRef index_size, co
 
   if (!iter.can_use_32bit_indexing()) {
     for (auto& sub_iter : iter.with_32bit_indexing()) {
-      gpu_index_kernel(sub_iter, index_size, index_stride, f);
+      gpu_index_kernel(sub_iter, index_size, index_stride, f, is_gather_like);
     }
     return;
   }
 
+
+  char* const out_ptr = static_cast<char*>(iter.data_ptr(0));
```

Comment:
Is data_ptr not a template method like it is for Tensors?

Reply:
no